---
title: "Project 2"
output: html_notebook
---

#Rubric Questions

How do we know that our question and hypothesis are adequate?
How many points if we can't get all of our code to run?
Do we have to explain why we can't get our code to run?
How do we know if we correctly analyzed the data?


#Introduction
Scientific Question: Is the amino acid sequences of the cancer target vascular endothelial growth factor receptor 2 (VEGFR2) less than 50% sequence identity to any other similar kinases thereby potentially explaining the high degree of selectivity of drugs against VEGFR2?

{5 points based on feedback from Project 2A and 2B}
VEGFR2 is a key gene involved in angiogenesis (Bowler 2019, Rahimi 2006), and is a receptor tyrosine kinase. Receptor tyrosine kinases consist of an extracellular ligand binding domain, a transmembrane helix, an intracellular juxtamembrane regulatory region that is often phosphorylated, a tyrosine kinase domain, and a carboxyl terminal tail (Hubbard 1998). As such, these proteins are very large and contain modular regions that are involved in different regulatory and signal transduction pathways in our cells (Du 2018).

There are a few known splice variants of VEGFR2. One splice variant, known as sVEGFR2, decreases lymphangiogenesis and downregulates neuroblastoma. Another splice variant, esVEGFR2, decreases lymphangiogenesis (Pavlakovic 2010). VEGFR2 is in the same family as VEGFR1 and VEGFR3 (Blume-Jensen, Nature 2001). Additionally, there are many other similar proteins in the human proteome, including the EGFR and ERBBR families of tyrosine kinases (BLAST search). These proteins are expressed differently than VEGFR2, which is highly expressed in endothelial cells (Liu 2018, Oncotarget). The question here is how similar are the regions of VEGFR2 involved in binding of small molecules that ultimately lead to inhibition (the kinase domain) versus the sequence similarity of the entire protein structure.

{5 points for description of background on the protein and citing your sources}
Scientific Hypothesis: If the VEGFR2 protein sequence is <70% similarity to off-target kinases, especially near the ligand binding site, then that would explain why there are not many adverse events in patients treated with VEGFR2 inhibitors.

{5 points based on feedback received in project 2A and 2B}
Here we perform multiple sequence alignment, pairwise sequence alignment and use a heatmap and sequence logo to visualize the similarity of the pairwise and multiple sequence alignment respectively. Multiple sequence alignment is done to be able to compare across all kinases for the regions specifically identified as in the ligand binding site. Bio3D is used to identify which residues are specifically found in the ligand binding site, and the structure for this analyses is read in from from the protein databank. Pairwise sequence alignment and heatmap analyses are done to answer the scientific question about 70% sequence identity (and hypothesis).

{10 points for description of work done to answer scientific question, and disprove/prove scientific hypothesis}

#Package Descriptions
Below, this bit of code is responsible for loading in all of the necessary packages. Here is where I will define the packages and what they are loaded in for

{10 points for adequately describing the packages}:
BioConductor:
Biostrings:
seqinr:
msa:
muscle:
Bio3D
Before running the code below, in the console I needed to install the packages using the commands:


```{r}
BiocManager::install("muscle")
library(BiocManager)
library(Biostrings)

Loading required package: BiocGenerics

 Attaching package: 'BiocGenerics'
 
The following objects are masked from 'package:stats':
   IQR, mad, sd, var, xtabs

 The following objects are masked from 'package:base':
  anyDuplicated, append, as.data.frame, basename, cbind, colnames,
  dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
 grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
 order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
    union, unique, unsplit, which.max, which.min
 
 Loading required package: S4Vectors
Warning: package 'S4Vectors' was built under R version 4.1.2
 Loading required package: stats4

 Attaching package: 'S4Vectors'
 
The following objects are masked from 'package:base':
     expand.grid, I, unname
 Loading required package: IRanges
Loading required package: XVector
 Loading required package: GenomeInfoDb

Warning: package 'GenomeInfoDb' was built under R version 4.1.2

 Attaching package: 'Biostrings'
 The following object is masked from 'package:base':

    strsplit
library(seqinr)
 
 Attaching package: 'seqinr'
The following object is masked from 'package:Biostrings':
 
    translate
library(msa)

Attaching package: 'msa'
 The following object is masked from 'package:BiocManager':
 
     version
```
Performing Bioinformatics Analyses
Loading in the Data from Uniprot download
Below, this code is responsible for loading in a fasta file for the VEGFR2 that is associated with EU826563.1 and many other similar kinases. This is just so that we can visualize the sequence in the R script.
```{r}
# Global variables are defined here that set up the rest of the script
# VEGFR2 unaligned is an AAStringSet variable or an S4 variable that is a global variable

VEGFR2_unaligned <- readAAStringSet("VEGFR2_seq.fasta")
# uncomment the line below to check what the datatype is
# typeof(VEGFR2_unaligned)

# VEGFR2_read is a list variable that is a global variable.
VEGFR2_read <- read.fasta(file = "VEGFR2_seq.fasta")
# uncomment the line below to check what the datatype is
#typeof(VEGFR2_read)

# VEGFR2_read is a list variable and so the length function can be used to determine the number of sequences
# in the list
length(VEGFR2_read)
## [1] 8
# Create a vector of data type integer that you can use to loop through each sequence to create the matrix
# for the heatplot.
x <- 1:length(VEGFR2_read)
# uncomment the line below to see what the vector x contains
#x
# uncomment the line below to see what the typeof vector is
#typeof(x)
```
The above fasta file was searched at UniProt for similar kinase sequences. The top 6 hits were all Homo Sapien variants that were compared here.

###Structural Bioinformatics In this next line of code, I will be using bio3d to identify which residues are in the ligand binding site. This will be used to later identify the similarities and differences between these residues in the alignment. To learn more about bio3d see the http://thegrantlab.org/bio3d_v2/html/binding.site.html

{2.5 points for adequately describing HOW bio3d identification of residues in the ligand binding site is done and how all the other methods in this project work }.
```{r}
library(bio3d)
## 
## Attaching package: 'bio3d'
## The following objects are masked from 'package:seqinr':
## 
##     consensus, read.fasta, write.fasta
## The following object is masked from 'package:Biostrings':
## 
##     mask
## The following object is masked from 'package:IRanges':
## 
##     trim
#install.packages("gplots")
library(gplots)
## 
## Attaching package: 'gplots'
## The following object is masked from 'package:IRanges':
## 
##     space
## The following object is masked from 'package:S4Vectors':
## 
##     space
## The following object is masked from 'package:stats':
## 
##     lowess
## Read in 2XIR pdb -> which is the structure of VEGFR2 with Pfizer compound PF- 00337210 bound to the kinase active site.
pdb <- read.pdb('2XIR')
##   Note: Accessing on-line PDB file
## Automatically identify what is the protein and what is the ligand
bs <- binding.site(pdb)

# Print the residues & save the resnumbers to a vector for later use
bs_resnumbers <- bs$resno

bs$resnames
##  [1] "LEU 840 (A)"  "VAL 848 (A)"  "ALA 866 (A)"  "LYS 868 (A)"  "GLU 885 (A)" 
##  [6] "LEU 889 (A)"  "VAL 899 (A)"  "VAL 914 (A)"  "VAL 916 (A)"  "GLU 917 (A)" 
## [11] "PHE 918 (A)"  "CYS 919 (A)"  "LYS 920 (A)"  "PHE 921 (A)"  "GLY 922 (A)" 
## [16] "LEU 1035 (A)" "CYS 1045 (A)" "ASP 1046 (A)" "PHE 1047 (A)"
```

```{r}
## Create a function (5 points) to create the matrix for the heatmap for any S4 sequences of any sequence length
## Could have also just read in the file and brought the above list and S4 definitions within the
## But having these definition outside of the function will also be helpful because now they are global variables
## instead of local variables. 

Matrix2Heatmap <- function(S4sequences, seq_length) {
# Prime the matrix
  forheatmap <- matrix(nrow=length(x), ncol=length(x))
  # The function creates a nested for loop
  for (i in seq_length) {
    for (j in seq_length){
      # define the variables for each sequence from the sequence alignment
      string1 <- S4sequences[i]
      string2 <- S4sequences[j]
    
      # Turn the string1 variable into a character string
      dsl = AAStringSetList(string1)
      as.character(unlist(dsl))
      cl = as(dsl, "CharacterList")
      as.list(cl)
      
      # Turn the string2 variable into a character string
      dsl2 = AAStringSetList(string2)
      as.character(unlist(dsl2))
      cl2 = as(dsl2, "CharacterList")
      as.list(cl2)
    
      # Perform a pairwise sequence alignment for the two strings
      pa <- pairwiseAlignment(pattern = c(cl2), subject = cl)
    
      # Assign the score from the pairwise sequence alignment to the matrix
      forheatmap[i,j] <- score(pa)
    }
  }
  return(forheatmap)
}

# Print the values in the heatmap
seq_matrix_align <- Matrix2Heatmap(VEGFR2_unaligned, x)

# Create the heatmap with a color key
heatmap.2(seq_matrix_align)
```


```{r}
# Multiple Sequence Alignment

system.file("tex", "texshade.sty", package="msa")
## [1] "/Library/Frameworks/R.framework/Versions/4.1/Resources/library/msa/tex/texshade.sty"
Kinase_MSA <- msa(VEGFR2_unaligned)
## use default substitution matrix
print(Kinase_MSA, show="complete")

```
The Below code will create a LaTeX object out of the sequences around 840-870 in the ligand binding site of the VEGFR kinases. In order for this code to work, you will need to intall tinytex which is the R package for managing LaTeX data. Then run the code: tinytex::parse_install(text = “! LaTeX Error: File `texshade.sty’ not found”) Once this code below prints out the latex file in your directory tree (Kinase_MSA.tex) then you will need toopen it in R by double clicking on it in the Files tab to the right. Once you do that click on Compile pdf. tinytex should enable the compiling of the tex file now.

We will create a Sequence Logo for residues in the binding site. A sequence Logo is a visual representation of a part of a sequence that is shared between homologs or similar proteins/genes. The size of each amino acid/nucleotide at each position is meant to represent how often that specific amino acid/nucleotide is found at that position. The X axis is numbered based on the reference gene/protein’s numbering scheme.
```{r}
# Code check below for what the first value is in the bs_resnumbers vector and how long that vector is.
bs_resnumbers[1]
## [1] 840
length(bs_resnumbers)
## [1] 19
# msaPrettyPrint is a function in the msa package. First you pass it the multiple sequence alignment output from
# above, Kinase_MSA. Then you specify the output type, and tex stands for latex. There are many other types you 
# could do but this is the one we are working with here so that we can create a PDF ultimately.
# Y specifies the residue sequence range to look at with the image. 
msaPrettyPrint(Kinase_MSA, output="tex", y=c((bs_resnumbers[1]+200), (bs_resnumbers[length(bs_resnumbers)]+200)),
subset=c(1:6), showNames="left", showLogo="top",
logoColors="rasmol", shadingMode="functional", shadingModeArg="structure",
showLegend=FALSE, askForOverwrite=FALSE)
```

Analysis of Results
Sequence 2 is the original full CDS. Sequences 1 and 3 are the VEGFR3 and VEGFR1. The other sequences are other similar protein tyrosine kinases. Based on the heatmap above, VEGFR2, VEGFR3, and EGFR are more similar to one another than they are to any of the other sequences, but they are still low sequence identify with a sequence alignment score less than 2000, suggesting lower than 50% sequence identify. This is in spite of the fact that a majority of the ligand binding site residues are conserved across the kinases as is evident from the Sequence Logo. The residues in the ligand binding site were identified by bio3d methods. Thus, my hypothesis was correct, the selectivity of many compounds for VEGFR2 is not necessarily due to the unique identity of the ligand binding site, but by some other mechanism, by which the overall sequence similarity is low compared to other kinases. This is not very intuitive, but has turned out to be correct. It is possible that these results are correlated but not causitive.
